.github/workflows/modules-zstd.yml

```diff
@@ -0,0 +1,58 @@
+name: Continuous Integration (ZSTD)
+on:
+  # Avoid triggering on pushes to /all/ open PR branches.
+  push:
+    branches:
+      - main
+    paths:
+      - 'xls/modules/zstd/**'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'xls/modules/zstd/**'
+  # This lets us trigger manually from the UI.
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Bazel Build (opt)
+    runs-on:
+      labels: ubuntu-22.04-64core
+    timeout-minutes: 600
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Restore Nightly Bazel Cache
+        uses: actions/cache/restore@v4
+        with:
+          path: "~/.cache/bazel"
+          key: bazel-cache-nightly-${{ runner.os }}-${{ github.sha }}
+          restore-keys: bazel-cache-nightly-${{ runner.os }}-
+
+      - name: Install dependencies via apt
+        run: sudo apt-get install python3-distutils python3-dev python-is-python3 libtinfo5  build-essential liblapack-dev libblas-dev gfortran
+
+      - name: Bazel Build Tools (opt)
+        run: |
+          bazel build -c opt --test_output=errors -- //xls/dslx:interpreter_main //xls/dslx/ir_convert:ir_converter_main //xls/tools:opt_main //xls/tools:codegen_main
+
+      - name: Build ZSTD Module (opt)
+        run: |
+          bazel test -c opt --test_output=errors -- //xls/modules/zstd:all
+
+      - name: Build ZSTD verilog targets (opt)
```

Comment:
curious if we want to attempt to run those even of the build fails? (I'd be fine with both)

Reply:
Added `continue-on-error: true` to specific steps of the workflow

Reply:
I think we want `continue-on-error` at the job level, so that we can get a red mark on the steps.

Reply:
Reorganized the workflow. There are multiple jobs defined now, each has `continue-on-error: true`.

There is:
* building and running tests of the ZSTD components,
* running IR benchmarks,
* verilog generation,
* synthesis,
* physical design flow

Testing job fails due to errors in the frame header fuzz test setup and 3 failing test cases for the top level proc. The rest of the jobs fail only due to the https://github.com/google/xls/issues/1455.

Reply:
Sorry for the miss-understanding. I didn't meant really to split those in different jobs but rather to move the `continue-on-error` attribute to the job level (rather than keeping it in on the individual steps).

One drawback on the current approach (separate jobs) is that you'll end up recompiling everything from scratch in each of them (and potentially trash out the cache, as all jobs are using the same key)

Reply:
@proppy I did that this way on purpose. Let me explain.

I first tried the exact approach you described here meaning single job with `continue-on-error: true` defined on job level with all the steps of the testing procedure one after another.
With that setup the effect was that the steps after the first failure in this job were still not being executed. You can see the results of this configuration in this [CI run](https://github.com/google/xls/actions/runs/9597434418/job/26466482211) based on [this](https://github.com/google/xls/actions/runs/9597434418/workflow) workflow (notice that the failure of `Test ZSTD Module (opt)` resulted in skipping the rest of the job)

That is why I separated each step into separate jobs. This gives the effect that you wanted (running the rest of the flow not minding the failures of e.g. DSLX tests). Having said that I do agree that most of the jobs will tend to recompile some stuff from scratch (tools, verilog generation, etc)

Please let me know how would you like to have this workflow organized.

Reply:
you're right `continue-on-error` seem to be related to matrix/workflow status: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error

Reply:
It seems to be a documented limitation of actions: https://github.com/actions/runner/issues/2347

Reply:
what about using `if: ${{ !cancelled() }}`  on indivual step that we want to run inconditionally as a workaround; https://docs.github.com/en/actions/learn-github-actions/expressions#always

Reply:
or maybe just revert to the original version (which stop the workflow on the first failure); as you prefer.

Reply:
`if: ${{ !cancelled() }}` does the trick. We can now have a single job for testing the whole ZSTD module and we are able to see the execution logs for each of the steps.

Reply:
CI looks good, thanks for working on this.